`middle-word-em=False` does not emit expected `strong`

**Describe the bug**

I'd like to use `middle-word-em` to remove random italics where `code-friendly` is not enough. One such case is `**worst_case_scenario**` which should emit `<p><strong>worst_case_scenario</strong></p>` but instead emits `<p>**worst_case_scenario**</p>`.

NB: it does not emit `strong` even if there's no `_` in the input.


**To Reproduce**

```python
from markdown2 import Markdown

def test_middle_word_em():
    md = Markdown(extras={"middle-word-em": False})
    assert md.convert("**worst_case_scenario**") == "<p><strong>worst_case_scenario</strong></p>"
```

You can verify the failure with `pytest -vvv`:

```
    def test_middle_word_em():
        md = Markdown(extras={"middle-word-em": False})
>       assert md.convert("**worst_case_scenario**") == "<p><strong>worst_case_scenario</strong></p>"
E       AssertionError: assert '<p>**worst_case_scenario**</p>\n' == '<p><strong>worst_case_scenario</strong></p>'
E         
E         - <p><strong>worst_case_scenario</strong></p>
E         + <p>**worst_case_scenario**</p>
```


**Expected behavior**
A clear and concise description of what you expected to happen.
`**worst_case_scenario**` should emit `<p><strong>worst_case_scenario</strong></p>`

**Debug info**
`markdown2==2.5.0`

Any extras being used:
`middle-word-em: False`

**Additional context**

it does not emit `strong` even if there's no `_` in the input.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -3,6 +3,7 @@
 ## python-markdown2 2.5.2 (not yet released)
 
 - [pull #605] Add support for Python 3.13, drop EOL 3.8
+- [pull #607] Fix `middle-word-em` extra preventing strongs from being recognized (#606)
 
 
 ## python-markdown2 2.5.1

--- a/lib/markdown2.py
+++ b/lib/markdown2.py
@@ -3144,7 +3144,20 @@ class MiddleWordEm(ItalicAndBoldProcessor):
 
         self.liberal_em_re = self.em_re
         if not options['allowed']:
-            self.em_re = re.compile(r'(?<=\b)%s(?=\b)' % self.liberal_em_re.pattern, self.liberal_em_re.flags)
+            self.em_re = re.compile(r'(?<=\b)%s(?=\b)' % self.em_re.pattern, self.em_re.flags)
+            self.liberal_em_re = re.compile(
+                r'''
+                    (                # \1 - must be a single em char in the middle of a word
+                        (?<![*_\s])  # cannot be preceeded by em character or whitespace (must be in middle of word)
+                        [*_]         # em character
+                        (?![*_])     # cannot be followed by another em char
+                    )
+                    (?=\S)           # em opening must be followed by non-whitespace text
+                    (.*?\S)          # the emphasized text
+                    \1               # closing char
+                    (?!\s|$)         # must not be followed by whitespace (middle of word) or EOF
+                '''
+                , re.S | re.X)
 
     def run(self, text):
         # run strong and whatnot first

--- a/lib/markdown2.py
+++ b/lib/markdown2.py
@@ -3160,6 +3160,10 @@ class MiddleWordEm(ItalicAndBoldProcessor):
                 , re.S | re.X)
 
     def run(self, text):
+        if self.options['allowed']:
+            # if middle word em is allowed, do nothing. This extra's only use is to prevent them
+            return text
+
         # run strong and whatnot first
         # this also will process all strict ems
         text = super().run(text)

--- a/test/tm-cases/middle_word_em_with_extra_ems.html
+++ b/test/tm-cases/middle_word_em_with_extra_ems.html
@@ -1,0 +1,19 @@
+<p>Double variants:</p>
+
+<p><strong>one_two_three</strong></p>
+
+<p><strong><em>one_two_three</em></strong></p>
+
+<p><em><strong>one_two_three</strong></em></p>
+
+<p><strong><em>one_two_three</em></strong></p>
+
+<p>Single variants:</p>
+
+<p><em>one_two_three</em></p>
+
+<p><em>one_two_three</em></p>
+
+<p><em>one*two*three</em></p>
+
+<p><em>one<em>two</em>three</em></p>

--- a/test/tm-cases/middle_word_em_with_extra_ems.opts
+++ b/test/tm-cases/middle_word_em_with_extra_ems.opts
@@ -1,0 +1,1 @@
+{"extras": {"middle-word-em": False}}

--- a/test/tm-cases/middle_word_em_with_extra_ems.text
+++ b/test/tm-cases/middle_word_em_with_extra_ems.text
@@ -1,0 +1,19 @@
+Double variants:
+
+**one_two_three**
+
+***one_two_three***
+
+_**one_two_three**_
+
+**_one_two_three_**
+
+Single variants:
+
+*one_two_three*
+
+_one_two_three_
+
+_one*two*three_
+
+*one*two*three*

